pkg/entities/configs.go

```diff
@@ -488,22 +488,20 @@ func (e *Entity) GetGuildRepostReactionConfigs(guildID string) ([]model.GuildCon
 	return e.repo.GuildConfigRepostReaction.GetByGuildID(guildID)
 }
 
-func (e *Entity) CreateRepostReactionEvent(req request.CreateMessageRepostHistRequest) (string, error) {
+func (e *Entity) CreateRepostReactionEvent(req request.CreateMessageRepostHistRequest) (*model.MessageRepostHistory, string, error) {
 	conf, err := e.repo.GuildConfigRepostReaction.GetByReaction(req.GuildID, req.Reaction)
 	if err != nil {
-		return "", err
+		return nil, "", err
 	}
 	if req.ReactionCount < conf.Quantity {
-		return "", nil
-	}
-	if isRepostable := e.IsRepostableMessage(req); !isRepostable {
-		return "", fmt.Errorf("message cannot be reposted")
+		return nil, "", nil
 	}
-	err = e.CreateRepostMessageHist(req, conf.RepostChannelID)
+
+	repostMsg, err := e.CreateRepostMessageHist(req, conf.RepostChannelID)
 	if err != nil {
-		return "", err
+		return nil, "", err
 	}
-	return conf.RepostChannelID, nil
+	return repostMsg, conf.RepostChannelID, nil
```

Comment:
why don't we set `repostMsg.RepostChannelID = conf.RepostChannelID` and `return respost, nil` only